docs/ci.md

```diff
@@ -0,0 +1,39 @@
+# Continuous Integration
+
+This documents describes the CI process for the solution - in partucular for the Spin apps.
+
+## Integration Tests
+
+The workflow [PR](../.github/workflows/PR.yaml) is used as a validation pipeline for pull requests created against the main branch.
+
+The job contains the generation and setup of the devcontainer including the docker-in-docker installation. That is required to build the wasm apps because it can be configured to use the containerd runtime for pulling and storing images as explaining in the [Docker-In-Docker](../docs/ci.md#docker-in-docker) section.
+The pipeline also configures the k3d cluster and deploys the apps into the cluster. The apps are validated by running the integration tests as part of the makefile experience.
+
+## Release Pipeline
+
+The workflow [Release](../.github/workflows/Release.yaml) is triggered when check-in into the main branch is commited (ideally as part of the PR merge) and is used to push the apps images into the GitHub Container Registry (GHCR).
```

Comment:
`when check-in into`  can be rephrased?

Reply:
done: when code is committed or merged into main branch